File: src/ProjectR/content/scripts/spells/BugStorm.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Bug Storm"
end

function GetDescription()
    return "Millions of bugs storm the enemy."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .66
end

function GetDelay()
  return .30
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  local damage = (4.5 * aAD * (aDRK/100) - 1.125 * dDEF) * (100/dDRK)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/BugStorm.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class BugStorm : SpellScriptBase
    {
        public override string Name { get { return "Bug Storm"; } }
        public override string Description { get { return "Millions of bugs storm the enemy."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 66; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (4.5 * cAD * (cDRK / 100) - 1.125 * tDEF) * (100 / tDRK);
            DealDamage(damage);
        }
    }
}